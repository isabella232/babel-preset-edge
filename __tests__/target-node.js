import buildPreset from "../src"

import { check, fixtures, titles } from "./core"

describe("Target: NodeJS", () => {
  const options = buildPreset(null, {
    sourceMaps: false,
    target: "node"
  })

  fixtures.forEach((fileName, index) => {
    test(titles[index], () => check(fileName, options))
  })
})
