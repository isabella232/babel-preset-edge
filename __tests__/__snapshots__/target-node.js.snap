// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Target: NodeJS alias 1`] = `
"\\"use strict\\";

require(\\"src/testonly\\");

var _Button = require(\\"../../src/components/button/Button\\"),
    _testonly = require(\\"../../src/testonly\\");

(0, _testonly.logx)(_Button.Button);"
`;

exports[`Target: NodeJS alias.tsx 1`] = `
"\\"use strict\\";

require(\\"src/testonly\\");

var _Button = require(\\"../../src/components/button/Button\\"),
    _testonly = require(\\"../../src/testonly\\");

(0, _testonly.logx)(_Button.Button);"
`;

exports[`Target: NodeJS comments 1`] = `
"\\"use strict\\";

var _interopRequireDefault = require(\\"@babel/runtime/helpers/interopRequireDefault\\"),
    _camelCase2 = _interopRequireDefault(require(\\"lodash/camelCase\\"));

function translate(value) {
  return (0, _camelCase2.default)(value);
}"
`;

exports[`Target: NodeJS deadcode 1`] = `
"\\"use strict\\";

const STATIC = false;
before();
after();"
`;

exports[`Target: NodeJS es2015: classes 1`] = `
"\\"use strict\\";

exports.__esModule = true;
exports.Component = void 0;

class Component extends Other {
  constructor() {
    super();
    this.flag = true;
  }

  main() {
    alert(this.foo);
  }

}

exports.Component = Component;"
`;

exports[`Target: NodeJS es2015: for_off 1`] = `
"\\"use strict\\";

require(\\"core-js/modules/es.array.iterator\\");

function test() {
  for (const taskName of tasks) {
    try {
      executeCommands();
    } catch (error) {
      console.error(\\"Error\\");
    }
  }
}"
`;

exports[`Target: NodeJS es2015: generator 1`] = `
"\\"use strict\\";

function* idMaker() {
  var index = 0;

  while (true) yield index++;
}"
`;

exports[`Target: NodeJS es2015: set 1`] = `
"\\"use strict\\";

require(\\"core-js/modules/es.array.iterator\\");

new Set([1, 2, 3]);"
`;

exports[`Target: NodeJS es2016: exponentiation 1`] = `
"\\"use strict\\";

var result = Math.pow(2, 3);"
`;

exports[`Target: NodeJS esnext: async 1`] = `
"\\"use strict\\";

require(\\"core-js/modules/es.promise\\");

function wait() {
  return new Promise(function ($return, $error) {
    return Promise.resolve(loadData()).then(function () {
      try {
        return $return();
      } catch ($boundEx) {
        return $error($boundEx);
      }
    }, $error);
  });
}"
`;

exports[`Target: NodeJS esnext: async_loop 1`] = `
"\\"use strict\\";

require(\\"core-js/modules/es.array.iterator\\");

require(\\"core-js/modules/es.promise\\");

function executeTasks(tasks) {
  return new Promise(function ($return, $error) {
    var $Loop_3_trampoline, $Loop_3_local;

    function $Loop_3_step() {
      var [taskName, $iterator_taskName_2] = $Loop_3_local();
      return $Loop_3.bind(this, taskName, $iterator_taskName_2);
    }

    function $Loop_3(taskName, $iterator_taskName_2) {
      $Loop_3_local = function () {
        return [taskName, $iterator_taskName_2];
      };

      if (!($iterator_taskName_2[1] = $iterator_taskName_2[0].next()).done && ((taskName = $iterator_taskName_2[1].value) || true)) {
        var $Try_1_Post = function () {
          try {
            return $Loop_3_step;
          } catch ($boundEx) {
            return $error($boundEx);
          }
        },
            $Try_1_Catch = function () {
          try {
            console.error(\\"Error\\");
            return $Try_1_Post();
          } catch ($boundEx) {
            return $error($boundEx);
          }
        };

        try {
          return Promise.resolve(executeCommands()).then(function () {
            try {
              return $Try_1_Post();
            } catch ($boundEx) {
              return $Try_1_Catch($boundEx);
            }
          }, $Try_1_Catch);
        } catch (error) {
          $Try_1_Catch(error)
        }
      } else return [1];
    }

    return ($Loop_3_trampoline = function (q) {
      while (q) {
        if (q.then) return void q.then($Loop_3_trampoline, $error);

        try {
          if (q.pop) {
            if (q.length) return q.pop() ? $Loop_3_exit.call(this) : q;else q = $Loop_3_step;
          } else q = q.call(this);
        } catch (_exception) {
          return $error(_exception);
        }
      }
    }.bind(this))($Loop_3.bind(this, void 0, [tasks[Symbol.iterator]()]));

    function $Loop_3_exit() {
      return $return();
    }
  });
}"
`;

exports[`Target: NodeJS esnext: catch_bind 1`] = `
"\\"use strict\\";

try {
  throw 0;
} catch (_unused) {
  doSomethingWhichDoesntCareAboutTheValueThrown();
} finally {
  doSomeCleanup();
}"
`;

exports[`Target: NodeJS esnext: class properties 1`] = `
"\\"use strict\\";

exports.__esModule = true;
exports.Component = void 0;

class Component {
  constructor() {
    this.onClick = () => {};
  }

}

exports.Component = Component;"
`;

exports[`Target: NodeJS esnext: decorators 1`] = `
"\\"use strict\\";

var _class;

let MyClass = annotation(_class = class MyClass {}) || _class;

function annotation(target) {
  target.annotated = true;
}"
`;

exports[`Target: NodeJS esnext: dynamic import 1`] = `
"\\"use strict\\";

require(\\"core-js/modules/es.promise\\");

import(
/*webpackChunkName:'lodash-wBYk8'*/
\\"lodash\\").then(lodash => {
  console.log(lodash.camelCase(\\"hello world\\"));
});"
`;

exports[`Target: NodeJS esnext: object spread 1`] = `
"\\"use strict\\";

let original = {
  bar: 10
},
    variant = Object.assign({
  foo: 1
}, original);"
`;

exports[`Target: NodeJS loadable 1`] = `
"\\"use strict\\";

var _interopRequireDefault = require(\\"@babel/runtime/helpers/interopRequireDefault\\");

require(\\"core-js/modules/es.promise\\");

exports.__esModule = true;
exports.Contact = exports.About = exports.Home = void 0;

var _loadableComponents = _interopRequireDefault(require(\\"loadable-components\\"));

const Home = (0, _loadableComponents.default)({
  chunkName() {
    return \\"Home-JfPYC\\";
  },

  isReady(props) {
    if (typeof __webpack_modules__ != 'undefined') {
      return !!__webpack_modules__[this.resolve(props)];
    }

    return false;
  },

  requireAsync: () => import(
  /* webpackChunkName: \\"Home-JfPYC\\" */
  \\"./views/Home\\"),

  requireSync(props) {
    const id = this.resolve(props);

    if (typeof __webpack_require__ != 'undefined') {
      return __webpack_require__(id);
    }

    return eval('module.require')(id);
  },

  resolve() {
    if (require.resolveWeak) {
      return require.resolveWeak(\\"./views/Home\\");
    }

    return eval('require.resolve')(\\"./views/Home\\");
  }

});
exports.Home = Home;
const About = (0, _loadableComponents.default)({
  chunkName() {
    return \\"About-8yDxR\\";
  },

  isReady(props) {
    if (typeof __webpack_modules__ != 'undefined') {
      return !!__webpack_modules__[this.resolve(props)];
    }

    return false;
  },

  requireAsync: () => import(
  /* webpackChunkName: \\"About-8yDxR\\" */
  \\"./views/About\\"),

  requireSync(props) {
    const id = this.resolve(props);

    if (typeof __webpack_require__ != 'undefined') {
      return __webpack_require__(id);
    }

    return eval('module.require')(id);
  },

  resolve() {
    if (require.resolveWeak) {
      return require.resolveWeak(\\"./views/About\\");
    }

    return eval('require.resolve')(\\"./views/About\\");
  }

});
exports.About = About;
const Contact = (0, _loadableComponents.default)({
  chunkName() {
    return \\"Contact-AIf6m\\";
  },

  isReady(props) {
    if (typeof __webpack_modules__ != 'undefined') {
      return !!__webpack_modules__[this.resolve(props)];
    }

    return false;
  },

  requireAsync: () => import(
  /* webpackChunkName: \\"Contact-AIf6m\\" */
  \\"./views/Contact\\"),

  requireSync(props) {
    const id = this.resolve(props);

    if (typeof __webpack_require__ != 'undefined') {
      return __webpack_require__(id);
    }

    return eval('module.require')(id);
  },

  resolve() {
    if (require.resolveWeak) {
      return require.resolveWeak(\\"./views/Contact\\");
    }

    return eval('require.resolve')(\\"./views/Contact\\");
  }

});
exports.Contact = Contact;"
`;

exports[`Target: NodeJS loadable magic-comments 1`] = `
"\\"use strict\\";

var _interopRequireDefault = require(\\"@babel/runtime/helpers/interopRequireDefault\\");

require(\\"core-js/modules/es.promise\\");

exports.__esModule = true;
exports.Contact = exports.About = exports.Home = void 0;

var _loadableComponents = _interopRequireDefault(require(\\"loadable-components\\"));

const Home = (0, _loadableComponents.default)({
  chunkName() {
    return \\"Home-JfPYC\\";
  },

  isReady(props) {
    if (typeof __webpack_modules__ != 'undefined') {
      return !!__webpack_modules__[this.resolve(props)];
    }

    return false;
  },

  requireAsync: () => import(
  /* webpackChunkName: \\"Home-JfPYC\\" */
  \\"./views/Home\\"),

  requireSync(props) {
    const id = this.resolve(props);

    if (typeof __webpack_require__ != 'undefined') {
      return __webpack_require__(id);
    }

    return eval('module.require')(id);
  },

  resolve() {
    if (require.resolveWeak) {
      return require.resolveWeak(\\"./views/Home\\");
    }

    return eval('require.resolve')(\\"./views/Home\\");
  }

});
exports.Home = Home;
const About = (0, _loadableComponents.default)({
  chunkName() {
    return \\"About\\";
  },

  isReady(props) {
    if (typeof __webpack_modules__ != 'undefined') {
      return !!__webpack_modules__[this.resolve(props)];
    }

    return false;
  },

  requireAsync: () => import(
  /* webpackChunkName: \\"About\\" */
  \\"./views/About\\"),

  requireSync(props) {
    const id = this.resolve(props);

    if (typeof __webpack_require__ != 'undefined') {
      return __webpack_require__(id);
    }

    return eval('module.require')(id);
  },

  resolve() {
    if (require.resolveWeak) {
      return require.resolveWeak(\\"./views/About\\");
    }

    return eval('require.resolve')(\\"./views/About\\");
  }

});
exports.About = About;
const Contact = (0, _loadableComponents.default)({
  chunkName() {
    return \\"Contact-AIf6m\\";
  },

  isReady(props) {
    if (typeof __webpack_modules__ != 'undefined') {
      return !!__webpack_modules__[this.resolve(props)];
    }

    return false;
  },

  requireAsync: () => import(
  /* webpackMode: \\"lazy\\", webpackFoo: \\"bar\\", webpackChunkName: \\"Contact-AIf6m\\" */
  \\"./views/Contact\\"),

  requireSync(props) {
    const id = this.resolve(props);

    if (typeof __webpack_require__ != 'undefined') {
      return __webpack_require__(id);
    }

    return eval('module.require')(id);
  },

  resolve() {
    if (require.resolveWeak) {
      return require.resolveWeak(\\"./views/Contact\\");
    }

    return eval('require.resolve')(\\"./views/Contact\\");
  }

});
exports.Contact = Contact;"
`;

exports[`Target: NodeJS loadable wildcard 1`] = `
"\\"use strict\\";

var _interopRequireDefault = require(\\"@babel/runtime/helpers/interopRequireDefault\\");

require(\\"core-js/modules/es.promise\\");

require(\\"core-js/modules/es.string.replace\\");

exports.__esModule = true;
exports.reactIntlData = exports.leanIntlData = exports.views = void 0;

var _loadableComponents = _interopRequireDefault(require(\\"loadable-components\\"));

const page = \\"About\\",
      lang = \\"de_DE\\",
      views = (0, _loadableComponents.default)({
  chunkName() {
    return \`views-\${page}\`.replace(/[^a-zA-Z0-9_!§$()=\\\\-^°]+/g, \\"-\\");
  },

  isReady(props) {
    if (typeof __webpack_modules__ != 'undefined') {
      return !!__webpack_modules__[this.resolve(props)];
    }

    return false;
  },

  requireAsync: () => import(
  /* webpackChunkName: \\"views-[request]\\" */
  \`./views/\${page}\`),

  requireSync(props) {
    const id = this.resolve(props);

    if (typeof __webpack_require__ != 'undefined') {
      return __webpack_require__(id);
    }

    return eval('module.require')(id);
  },

  resolve() {
    if (require.resolveWeak) {
      return require.resolveWeak(\`./views/\${page}\`);
    }

    return eval('require.resolve')(\`./views/\${page}\`);
  }

});
exports.views = views;
const leanIntlData = (0, _loadableComponents.default)({
  chunkName() {
    return \`lean-intl-locale-data-\${lang}\`.replace(/[^a-zA-Z0-9_!§$()=\\\\-^°]+/g, \\"-\\");
  },

  isReady(props) {
    if (typeof __webpack_modules__ != 'undefined') {
      return !!__webpack_modules__[this.resolve(props)];
    }

    return false;
  },

  requireAsync: () => import(
  /* webpackChunkName: \\"lean-intl-locale-data-[request]\\" */
  \`./lean-intl/locale-data/\${lang}\`),

  requireSync(props) {
    const id = this.resolve(props);

    if (typeof __webpack_require__ != 'undefined') {
      return __webpack_require__(id);
    }

    return eval('module.require')(id);
  },

  resolve() {
    if (require.resolveWeak) {
      return require.resolveWeak(\`./lean-intl/locale-data/\${lang}\`);
    }

    return eval('require.resolve')(\`./lean-intl/locale-data/\${lang}\`);
  }

});
exports.leanIntlData = leanIntlData;
const reactIntlData = (0, _loadableComponents.default)({
  chunkName() {
    return \`react-intl-locale-data-\${lang}\`.replace(/[^a-zA-Z0-9_!§$()=\\\\-^°]+/g, \\"-\\");
  },

  isReady(props) {
    if (typeof __webpack_modules__ != 'undefined') {
      return !!__webpack_modules__[this.resolve(props)];
    }

    return false;
  },

  requireAsync: () => import(
  /* webpackChunkName: \\"react-intl-locale-data-[request]\\" */
  \`./react-intl/locale-data/\${lang}\`),

  requireSync(props) {
    const id = this.resolve(props);

    if (typeof __webpack_require__ != 'undefined') {
      return __webpack_require__(id);
    }

    return eval('module.require')(id);
  },

  resolve() {
    if (require.resolveWeak) {
      return require.resolveWeak(\`./react-intl/locale-data/\${lang}\`);
    }

    return eval('require.resolve')(\`./react-intl/locale-data/\${lang}\`);
  }

});
exports.reactIntlData = reactIntlData;"
`;

exports[`Target: NodeJS lodash: async 1`] = `
"\\"use strict\\";

var _interopRequireDefault = require(\\"@babel/runtime/helpers/interopRequireDefault\\"),
    _parallel2 = _interopRequireDefault(require(\\"async/parallel\\"));

(0, _parallel2.default)([]);"
`;

exports[`Target: NodeJS lodash: lodash 1`] = `
"\\"use strict\\";

var _interopRequireDefault = require(\\"@babel/runtime/helpers/interopRequireDefault\\"),
    _camelCase2 = _interopRequireDefault(require(\\"lodash/camelCase\\"));

(0, _camelCase2.default)(\\"hello world\\");"
`;

exports[`Target: NodeJS lodash: recompose 1`] = `
"\\"use strict\\";

var _interopRequireDefault = require(\\"@babel/runtime/helpers/interopRequireDefault\\"),
    _compose2 = _interopRequireDefault(require(\\"recompose/compose\\"));

(0, _compose2.default)([]);"
`;

exports[`Target: NodeJS macro: gql 1`] = `
"\\"use strict\\";

const query = {
  kind: \\"Document\\",
  definitions: [{
    kind: \\"OperationDefinition\\",
    operation: \\"query\\",
    name: {
      kind: \\"Name\\",
      value: \\"User\\"
    },
    variableDefinitions: [],
    directives: [],
    selectionSet: {
      kind: \\"SelectionSet\\",
      selections: [{
        kind: \\"Field\\",
        name: {
          kind: \\"Name\\",
          value: \\"user\\"
        },
        arguments: [{
          kind: \\"Argument\\",
          name: {
            kind: \\"Name\\",
            value: \\"id\\"
          },
          value: {
            kind: \\"IntValue\\",
            value: \\"5\\"
          }
        }],
        directives: [],
        selectionSet: {
          kind: \\"SelectionSet\\",
          selections: [{
            kind: \\"Field\\",
            name: {
              kind: \\"Name\\",
              value: \\"lastName\\"
            },
            arguments: [],
            directives: []
          }, {
            kind: \\"FragmentSpread\\",
            name: {
              kind: \\"Name\\",
              value: \\"UserEntry1\\"
            },
            directives: []
          }]
        }
      }]
    }
  }],
  loc: {
    start: 0,
    end: 79,
    source: {
      body: \\"\\\\n  query User {\\\\n    user(id: 5) {\\\\n      lastName\\\\n      ...UserEntry1\\\\n    }\\\\n  }\\\\n\\",
      name: \\"GraphQL request\\",
      locationOffset: {
        line: 1,
        column: 1
      }
    }
  }
};"
`;

exports[`Target: NodeJS macro: preval 1`] = `
"\\"use strict\\";

const one = 1;"
`;

exports[`Target: NodeJS promise 1`] = `
"\\"use strict\\";

require(\\"core-js/modules/es.promise\\");

exports.__esModule = true;
exports.default = _default;

function _default() {
  return new Promise(resolve => {
    return resolve(42);
  });
}"
`;

exports[`Target: NodeJS react: jsx 1`] = `
"\\"use strict\\";

function render() {
  return React.createElement(\\"h1\\", null, \\"Hello\\");
}"
`;

exports[`Target: NodeJS react: jsx spread 1`] = `
"\\"use strict\\";

function render(props) {
  return React.createElement(\\"h1\\", Object.assign({
    className: \\"large\\"
  }, props), \\"Hello\\");
}"
`;

exports[`Target: NodeJS require context 1`] = `
"\\"use strict\\";

if (process.env.NODE_ENV === \\"test\\") {
  require('babel-plugin-require-context-hook/register')();
}

const loader = __requireContext(__dirname, __dirname, false, /\\\\.js/);"
`;

exports[`Target: NodeJS typescript.tsx 1`] = `
"\\"use strict\\";

exports.__esModule = true;
exports.hasPackageConfig = void 0;

const hasPackageConfig = name => {
  try {
    return !!getPackageConfig()[name];
  } catch (_) {
    return false;
  }
};

exports.hasPackageConfig = hasPackageConfig;"
`;
